### PR TITLE
DO NOT MERGE - Expose max channel size in bytes via kafka connector

### PR DESCRIPTION
**Pull request by sfc-gh-kgaputis**
# Overview

SNOW-XXXXX

## Pre-review checklist
- [ ] This change should be part of a Behavior Change Release. See [go/behavior-change](http://go/behavior-change).
- [ ] This change has passed Merge gate tests
- [ ] Snowpipe Changes
- [ ] Snowpipe Streaming Changes
- [ ] This change is TEST-ONLY
- [ ] This change is README/Javadocs only
- [ ] This change is protected by a config parameter <PARAMETER_NAME> eg `snowflake.ingestion.method`.
    - [ ] `Yes` - Added end to end and Unit Tests. 
    - [ ] `No` - Suggest why it is not param protected
- [ ] Is his change protected by parameter <PARAMETER_NAME> on the server side?
    - [ ] The parameter/feature is not yet active in production (partial rollout or PrPr, see [Changes for Unreleased Features and Fixes](http://go/ppp-prpr)).
    - [ ] If there is an issue, it can be safely mitigated by turning the parameter off. This is also verified by a test (See [go/ppp](http://go/ppp)).

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java [hunk 1]
```diff
@@ -121,6 +121,10 @@ public class SnowflakeSinkConnectorConfig {
   public static final String SNOWPIPE_STREAMING_MAX_CLIENT_LAG =
       "snowflake.streaming.max.client.lag";
 
+  // This is the streaming max client lag which can be defined in config
```
Comment (sfc-gh-tjones) [inline src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java:124]: Nit, `streaming max channel size (in bytes)`

**Review by sfc-gh-japatel — COMMENTED** [review]
hey @sfc-gh-kgaputis thanks a lot for contribution
we would rather want to make this a generic config. 

https://snowflakecomputing.atlassian.net/browse/SNOW-1117446

TLDR:
instead of new KC config for every new/future config in streaming client, we can just use a key value pair 
"snowflake.streaming.client.provider.properties.override":"max_client_lag:5000,key:value"


Key should be present in https://github.com/snowflakedb/snowflake-ingest-java/blob/master/src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

File: src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java [hunk 2]
```diff
@@ -551,6 +555,17 @@ static ConfigDef newConfigDef() {
...
             SNOWPIPE_STREAMING_MAX_CLIENT_LAG)
+        .define(
+            SNOWPIPE_STREAMING_MAX_CHANNEL_SIZE,
+            Type.LONG,
+            StreamingUtils.STREAMING_BUFFER_MAX_CHANNEL_SIZE_DEFAULT_BYTES,
+            ConfigDef.Range.atLeast(StreamingUtils.STREAMING_BUFFER_MAX_CHANNEL_SIZE_MINIMUM_BYTES),
+            Importance.LOW,
+            "Specify the maximum size (in bytes) of a single channel flush performed by the Ingest SDK",
+            CONNECTOR_CONFIG,
+            7, //TODO verify what this should be
```
Comment (sfc-gh-tzhang) [inline src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java:566]: I'm not sure how important this is, but there is a 7 below already, see ENABLE_STREAMING_CLIENT_OPTIMIZATION_CONFIG, you can put it in the end and use a new number just in case

File: src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingUtils.java [hunk 3]
```diff
@@ -45,6 +46,14 @@ public class StreamingUtils {
   public static final long STREAMING_BUFFER_FLUSH_TIME_DEFAULT_SEC =
       Duration.ofSeconds(10).getSeconds();
 
+
+  //TODO decide if this makes sense
```
Comment (sfc-gh-tzhang) [inline src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingUtils.java:50]: Yeah, let's do 32MB

**Review by sfc-gh-tzhang — APPROVED** [review]
LGTM, thanks! Please do a quick test with your change to make sure the file size do increase, you can check the file size in the log

**Comment by sfc-gh-japatel** [comment]
I am working on a PR for a map of key value pair, please hold on to this PR, but we appreciate the contribution!

**Review by sfc-gh-japatel — CHANGES_REQUESTED** [review]
left a comment!

File: src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java (hunk 1 above)
Comment (sfc-gh-kgaputis) [reply src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java:124]: The name seemed a bit long but I changed config to `snowflake.streaming.max.channel.size.in.bytes` to be more explicit.

File: src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java (hunk 2 above)
Comment (sfc-gh-kgaputis) [reply src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java]: Updated to 10, which was unused in the `CONNECTOR_CONFIG` group.

**Comment by sfc-gh-kgaputis** [comment]
Thank you for reviewing team.   I like the plan @sfc-gh-japatel  and @sfc-gh-tzhang to make all the low level SDK settings overridable from a single param.
